components/shader/shadermanager.cpp

```diff
@@ -145,7 +146,7 @@ namespace Shader
             static unsigned int counter = 0;
             shader->setName(std::to_string(counter++));
 
-            shaderIt = mShaders.insert(std::make_pair(std::make_pair(shaderTemplate, defines), shader)).first;
+            shaderIt = mShaders.insert(std::make_pair(std::make_pair(shaderTemplate, localdefines), shader)).first;
         }
```

Comment:
This won't work if the global defines are changed while OpenMW is running (which you should be ready for them to do as we have an in-game settings menu).

Reply:
What do you think of a int lock on globaldefines (preventing it's mod during run)?

Reply:
A lock alone won't help with this in any way, shape or form. Take a look at the implementation in my PR.

Reply:
I see what you did, i just made a simplier version that could be incrementally updated...
Further the feature of changing shaders global define during prog run should be for further pr: I guess it would be like adding gui shadow conf for the first pr..Not now
For that, I purpose to add an int guard to prevent globaldefine mod after shadowmanager updated them...
No?